ocs_ci/utility/rosa.py

```diff
@@ -396,7 +424,9 @@ def install_odf_addon(cluster):
         public_key_only = remove_header_footer_from_key(public_key)
         cmd += f' --onboarding-validation-key "{public_key_only}"'
 
-    if cluster_type.lower() == "consumer" and provider_name:
+    if cluster_type.lower() == "consumer" and (
+        provider_name or config.ENV_DATA.get("appliance_mode", False)
+    ):
```

Comment:
Can you please provide more info on this if statement and maybe add a note? The statement would be true if cluster type is consumer, appliance mode is false and provider name is not set.

Reply:
The condition is true when Cluster type is consumer and provider name given
or 
The condition is true when Cluster type is consumer and appliance_mode is True

Reply:
Right, I am just a little confused because it is possible to have this condition true when provider name is not set and in the following code is provider name used. It is probably ok because the function `get_storage_provider_endpoint` actually is not using it currently. I am not sure if it's ok to make the condition true if consumer is appliance mode without having set provider name.

Reply:
Verification Job for addon mode consumer deployment  where provider name is given and appliance_mode=false
https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/13660/

>>if the consumer is appliance mode without having set provider name.==> Yes , in appliance mode provider name is neither used for getting subnet-ids nor for getting endpoint. 
Verification Job in https://github.com/red-hat-storage/ocs-ci/pull/6001#issuecomment-1150984671